components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,255 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
+		host := getHostFromNodeTypeAndIpCombination(nodeTypeWithIp)
+		startMockServerRequestBody := models.MockServerRequestBody{}
+		if len(protocolMap[constants.TCP]) != 0 {
+			startMockServerRequestBody.Protocol = constants.TCP
+			ss.constructRequestForStartingMockServer(protocolMap, host, startMockServerRequestBody, startMockServerChannel, &ipPortProtocolMap, &totalReq)
+		}
+		if len(protocolMap[constants.UDP]) != 0 {
+			startMockServerRequestBody.Protocol = constants.UDP
+			ss.constructRequestForStartingMockServer(protocolMap, host, startMockServerRequestBody, startMockServerChannel, &ipPortProtocolMap, &totalReq)
+		}
+		if len(protocolMap[constants.HTTP]) != 0 {
+			startMockServerRequestBody.Protocol = constants.HTTP
+			ss.constructRequestForStartingMockServer(protocolMap, host, startMockServerRequestBody, startMockServerChannel, &ipPortProtocolMap, &totalReq)
+		}
+		if len(protocolMap[constants.HTTPS]) != 0 {
+			startMockServerRequestBody.Protocol = constants.HTTPS
+			ss.generateRootCaAndPrivateKeyForHost(host, &startMockServerRequestBody)
+			ss.constructRequestForStartingMockServer(protocolMap, host, startMockServerRequestBody, startMockServerChannel, &ipPortProtocolMap, &totalReq)
+		}
+	}
+
+	successfullyStartedMockServers := []models.MockServerFromBatchServiceResponse{}
+	notStartedMockServers := []models.MockServerFromBatchServiceResponse{}
+	for i := 0; i < totalReq; i++ {
+		result := <-startMockServerChannel
+		if result.Error == nil {
+			successfullyStartedMockServers = append(successfullyStartedMockServers, result)
+		} else {
+			notStartedMockServers = append(notStartedMockServers, result)
+		}
+
+	}
+	defer close(startMockServerChannel)
+	return successfullyStartedMockServers, notStartedMockServers
+}
+
+func (ss *BatchCheckService) constructRequestForStartingMockServer(protocolMap map[string][]int, host string, startMockServerRequestBody models.MockServerRequestBody, startMockServerChannel chan models.MockServerFromBatchServiceResponse, ipPortProtocolMap *map[string]bool, totalReq *int) {
+	protocolType := startMockServerRequestBody.Protocol
+	for _, port := range protocolMap[protocolType] {
+		startMockServerRequestBody.Port = port
+		key := fmt.Sprint(host, "_", protocolType, "_", port)
+
+		if !(*ipPortProtocolMap)[key] {
+			go ss.startMockServerOnHostAndPort(host, ss.port, startMockServerRequestBody, startMockServerChannel)
+			*totalReq = *totalReq + 1
+			(*ipPortProtocolMap)[key] = true
+		}
+	}
+}
+
+func getHostFromNodeTypeAndIpCombination(nodeTypeWithIp string) string {
+	return nodeTypeWithIp[strings.Index(nodeTypeWithIp, "_")+1:]
+}
+
+func (ss *BatchCheckService) generateRootCaAndPrivateKeyForHost(host string, startMockServerRequestBody *models.MockServerRequestBody) {
+	err := certgenerateutils.GenerateCert(host)
+	if err != nil {
+		ss.log.Error(err)
+		return
+	}
+	certificateFileName := "certificate.pem"
+	privateKeyFileName := "private_key.pem"
```

Comment:
```suggestion
	certificateFileName, privateKeyFileName, err := certgenerateutils.GenerateCert(host)
	if err != nil {
		ss.log.Error(err)
		return
	}
	defer ss.fileUtils.DeleteFile(certificateFileName)
	defer ss.fileUtils.DeleteFile(privateKeyFileName)
	
```
Please use defer like this , its not required to defer same function multiple times and we want to delete the tempporary files as soon as function returns

Reply:
Done